qiskit/qpy/binary_io/circuits.py

```diff
@@ -578,7 +580,11 @@ def _dumps_instruction_parameter(param, index_map, use_symengine):
 
 # pylint: disable=too-many-boolean-expressions
 def _write_instruction(file_obj, instruction, custom_operations, index_map, use_symengine):
-    gate_class_name = instruction.operation.base_class.__name__
+    gate_class = getattr(instruction.operation, "base_class", None)
```

Comment:
Would it also be valid to type-check if `instruction` is of type `Instruction` and hence has a `base_class`?